src/NuGet.Core/NuGet.Packaging/Signing/Utility/CertificateChainUtility.cs

```diff
@@ -212,27 +212,27 @@ internal static bool ChainStatusListIncludesStatus(X509ChainStatus[] chainStatus
             return chainStatus.Any();
         }
 
-        internal static bool TryGetStatusMessage(X509ChainStatus[] chainStatuses, X509ChainStatusFlags status, out IEnumerable<string> messages)
+        internal static bool TryGetStatusAndMessage(X509ChainStatus[] chainStatuses, X509ChainStatusFlags status, out IEnumerable<string> statusAndMessages)
         {
-            messages = null;
+            statusAndMessages = null;
 
             if (ChainStatusListIncludesStatus(chainStatuses, status, out var chainStatus))
             {
-                messages = GetMessagesFromChainStatuses(chainStatus);
+                statusAndMessages = GetStatusAndMessagesFromChainStatuses(chainStatus);
 
                 return true;
             }
 
             return false;
         }
 
-        internal static IEnumerable<string> GetMessagesFromChainStatuses(IEnumerable<X509ChainStatus> chainStatuses)
+        internal static IEnumerable<string> GetStatusAndMessagesFromChainStatuses(IEnumerable<X509ChainStatus> chainStatuses)
         {
             return chainStatuses
-                .Select(x => x.StatusInformation?.Trim())
+                .Select(x => $"{x.Status}: {x.StatusInformation?.Trim()}")
```

Comment:
What would happen if x.StatusInformation.Trim() returns an empty string ? With the previous code, it seems x will be removed from final results. With this change, x would be included.

Reply:
Thanks Fernando! Fixed.